test/functional/test_framework/wallet.py

```
@@ -185,9 +223,40 @@
+       corresponding public key, scriptPubKey and address. Supported types are
+       'legacy'. Can be used when a random destination is needed, but no
+       compiled wallet is available (e.g. as replacement to the
+       getnewaddress/getaddressinfo RPCs)."""
+    key = ECKey()
+    key.generate()
+    pubkey = key.get_pubkey().get_bytes()
+    if address_type == 'legacy':
+        scriptpubkey = key_to_p2pkh_script(pubkey)
+        address = key_to_p2pkh(pubkey)
+    else:
+        assert False
```

Ruff (0.8.2)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)